lib/ptracer.h

```diff
@@ -0,0 +1,485 @@
+    lua_getfield(L, LUA_REGISTRYINDEX, PALLENE_TRACER_CONTAINER_ENTRY);
+    pt_cont_t *cont = (pt_cont_t *) lua_touserdata(L, -1);
+    pt_frame_t *stack = cont->stack;
+    /* The point where we are in the Pallene stack. */
+    int index = cont->count - 1;
+    lua_pop(L, 1);
+
+    /* Max number of white and black frames. */
+    int mwhite, mblack;
+    _pallene_tracer_countframes(cont, &mwhite, &mblack);
+    /* Max levels of Lua stack. */
+    int mlevel = _pallene_tracer_countlevels(L);
+
+    /* Total frames we are going to print. */
+    /* Black frames are used for switching and we will start from
+       Lua stack level 1. */
+    int nframes = mlevel + mwhite - mblack - 1;
+    /* Amount of frames printed. */
+    int pframes = 0;
+    /* Should we print ellipsis? */
+    bool ellipsis = nframes > (PALLENE_TRACEBACK_TOP_THRESHOLD
+        + PALLENE_TRACEBACK_BOTTOM_THRESHOLD);
+
+    /* Buffer to store for a single frame line to be printed. */
+    char buf[1024];
+
+    const char *message = lua_tostring(L, 1);
+    fprintf(stderr, "Runtime error: %s\nStack traceback:\n", message);
+
+    lua_Debug ar;
+    int top = lua_gettop(L);
+    int level = 1;
+
+    while(lua_getstack(L, level++, &ar)) {
+        /* Get additional information regarding the frame. */
+        lua_getinfo(L, "Slntf", &ar);
+
+        /* If the frame is a C frame. */
+        if(lua_iscfunction(L, -1)) {
+            if(index >= 0) {
+                /* Check whether this frame is tracked (Pallene C frames). */
+                int check = index;
+                while(stack[check].type != PALLENE_TRACER_FRAME_TYPE_LUA)
+                    check--;
+
+                /* If the frame signature matches, we switch to printing Pallene frames. */
+                if(lua_tocfunction(L, -1) == stack[check].shared.frame_sig) {
+                    /* Now print all the frames in Pallene stack. */
+                    for(; index > check; index--) {
+                        snprintf(buf, 1023, "    %s:%d: in function '%s'\n",
+                            stack[index].shared.details->filename,
+                            stack[index].line, stack[index].shared.details->fn_name);
+                        DBG_PRINT();
+                    }
+
+                    /* 'check' idx is guaranteed to be a Lua interface frame.
+                       Which is basically our 'stack' index at this point. So,
+                       we simply ignore the Lua interface frame. */
+                    index--;
+
+                    /* We are done. */
+                    lua_settop(L, top);
+                    continue;
+                }
+            }
+
+            /* Then it's an untracked C frame. */
+            if(_pallene_tracer_pgf_name(L))
+                lua_pushfstring(L, "%s", lua_tostring(L, -1));
+            else lua_pushliteral(L, "<?>");
+
+            snprintf(buf, 1023, "    C: in function '%s'\n", lua_tostring(L, -1));
```

Comment:
I know its very unlikely that a function name can exceed `1023`,
But we should add a sanity check here, at least in debug mode:

```c
const int maxbufsize = vsnprintf(NULL, 0, "...", ...);
assert(maxbufsize <= 1024);
```

Not very important, but worth noting for defensive programming.